### ISSUE TITLE
Question: about the future

### ISSUE THREAD

**Issue by 7jrxt42BxFZo4iAnN4CX** [wvietor/ibkr_rust#1]
How do you do? I would like to ask if there are plans to implement the request. And how long would that take?

I just started studying rust and I don't know if I can help yet. But I'm interested in the idea of having rust API instead of Java.

**Comment by wvietor** [comment]
Yeah, I am missing some client functionality. And yes, the existing rust crates aren’t great. A lot of them don’t really take advantage of rust’s type system and just port some of the bloated IBKR classes to rust. My goal with this crate was to try to encode as many of the API requirements into the types themselves — eg. IBKR doesn’t have “TRADES” historical bars for forex, so you’ll get a compiler error if you try to do this.

After I finish the new error handling, I have to write some wrapper functions to finish handling some existing requests, but I will work on adding the other client methods. The only functionality I wasn’t planning to implement was the “combo legs,” bond-related requests, and the fundamental/news requests. 

Also, I have written a fair bit of documentation. You can access it by running “cargo doc” from the crate directory. It outputs a bunch of html in the target/doc directory, which you can look at from the browser just like any ordinary crate docs. This should make it easier to understand the functions without having to read all the source code. I think there’s some info about the docs in the rust book too if you have more questions.

**Comment by wvietor** [comment]
Hey! I think tracking the limits is a good idea. Saw your commit -- will merge it shortly.
